solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/test/java/com/solace/spring/cloud/stream/binder/provisioning/SolaceProvisioningUtilQueueNameTest.java

```diff
@@ -212,4 +217,155 @@ private SolaceConsumerProperties createConsumerProperties(String prefix, boolean
+        consumerProperties.getExtension().setQueueNamePrefix("    aQueueNamePrefixWithSpaces    ");
+
+        String destination = "  destination/with/spaces      ";
+        String group = "    aGroupWithSpaces    ";
+
+        SolaceProvisioningUtil.QueueNames queueNames = SolaceProvisioningUtil.getQueueNames(destination, group, consumerProperties,
+                SolaceProvisioningUtil.isAnonQueue(group));
+
+        assertEquals("aQueueNamePrefixWithSpaces/wk/aGroupWithSpaces/plain/destination/with/spaces", queueNames.getConsumerGroupQueueName());
+        assertEquals("aQueueNamePrefixWithSpaces/error/wk/aGroupWithSpaces/plain/destination/with/spaces", queueNames.getErrorQueueName());
+    }
+
+    @Test
+    public void testDefaultQueueNameExpressionsWithGroupAsWhiteSpacesOnlyGeneratesAGroup() {
+        ExtendedConsumerProperties<SolaceConsumerProperties> consumerProperties = new ExtendedConsumerProperties<>(new SolaceConsumerProperties());
+
+        String destination = "simple/destination";
+        String group = "    ";
+
+        SolaceProvisioningUtil.QueueNames queueNames = SolaceProvisioningUtil
+                .getQueueNames(destination, group, consumerProperties, SolaceProvisioningUtil.isAnonQueue(group));
+
+        assertThat(queueNames.getConsumerGroupQueueName(), matchesRegex("scst\\/an\\/\\b[0-9a-f]{8}\\b(?:-[0-9a-f]{4}){3}-\\b[0-9a-f]{12}\\b\\/plain\\/simple\\/destination"));
+        assertThat(queueNames.getErrorQueueName(), matchesRegex("scst\\/error\\/an\\/\\b[0-9a-f]{8}\\b(?:-[0-9a-f]{4}){3}-\\b[0-9a-f]{12}\\b\\/plain\\/simple\\/destination"));
+    }
+
+    @Test
+    public void testDefaultQueueNameExpressionsWithGroupAsNullGeneratesAGroup() {
+        ExtendedConsumerProperties<SolaceConsumerProperties> consumerProperties = new ExtendedConsumerProperties<>(new SolaceConsumerProperties());
+
+        String destination = "simple/destination";
+        String group = null;
+
+        SolaceProvisioningUtil.QueueNames queueNames = SolaceProvisioningUtil
+                .getQueueNames(destination, group, consumerProperties, SolaceProvisioningUtil.isAnonQueue(group));
+
+        assertThat(queueNames.getConsumerGroupQueueName(), matchesRegex("scst\\/an\\/\\b[0-9a-f]{8}\\b(?:-[0-9a-f]{4}){3}-\\b[0-9a-f]{12}\\b\\/plain\\/simple\\/destination"));
+        assertThat(queueNames.getErrorQueueName(), matchesRegex("scst\\/error\\/an\\/\\b[0-9a-f]{8}\\b(?:-[0-9a-f]{4}){3}-\\b[0-9a-f]{12}\\b\\/plain\\/simple\\/destination"));
+    }
+
+    @Test
+    public void testQueueNameExpressionWithStaticValue() {
+        SolaceConsumerProperties consumerProperties = createConsumerProperties(null, true, true, true, true);
+        //The escaped single quote '' resolves to a single quote in the actual queue name
+        consumerProperties.setQueueNameExpression("'My/Static.QueueName-_''>*!@#$%^&()+='");
```

Comment:
Add `#{test}:[]{}|\\\"-~` to this string as well. This should render as-is like the other special characters.